### PR TITLE
Update for new grpc api

### PR DESCRIPTION
**Pull request by squidboylan**
**What type of PR is this?**

> /kind feature

**Any specific area of the project related to this PR?**

> /area client

**What this PR does / why we need it**:

Adds support for the new grpc api

**Does this PR introduce a user-facing change?**:

Requires changes to use the updated grpc api

### CODE REVIEW

**Comment by fntlnz** [comment]
@squidboylan thanks for taking care of this! 
Would you mind adding an example for the bidirectional use case?
The example we have also should not have the `loop` anymore since it can't really do anything after it receives.

**Review by leodido — COMMENTED** [review]
I tried it and was not able to make it work correctly.

By looking briefly at the code I believe that should be two routines: one that sends requests and the other that (eventually) receives the Falco alerts (because of the bidirectional streaming nature of the `sub` API).

**Comment by nibalizer** [comment]
Thanks @squidboylan for working on this PR! We appreciate it and we hope to have good rust bindings to enable collaboration with that entire space.

**Comment by squidboylan** [comment]
> I tried it and was not able to make it work correctly.
> 
> By looking briefly at the code I believe that should be two routines: one that sends requests and the other that (eventually) receives the Falco alerts (because of the bidirectional streaming nature of the `sub` API).

Ok this should be implemented and work as expected, let me know if you see anything else

File: Cargo.toml [hunk 1]
```diff
@@ -13,12 +13,12 @@ keywords = ["falco", "falcosecurity", "kubernetes", "security"]
 [dependencies]
 grpcio = { version = "0.4.7", features = ["openssl"] }
 failure = "0.1"
-futures = "^0.1.15"
-protobuf = "~2"
+futures = "^0.1.29"
+protobuf = "~2.17"
```
Comment (leodido) [inline Cargo.toml:17]:
```
error[E0658]: non-builtin inner attributes are unstable                                                               │
 --> /home/vagrant/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.17.0/src/well_known_types/type_pb.rs:9:1│
  |                                                                                                                   │
9 | #![rustfmt::skip]                                                                                                 │
  | ^^^^^^^^^^^^^^^^^                                                                                                 │
  |                                                                                                                   │
  = note: for more information, see https://github.com/rust-lang/rust/issues/54726
```

It seems to me that protobuf 2.17 contains unstable features (that only works with an unstable `rustc`).

Comment (leodido) [reply Cargo.toml:17]: Was this version bump mandatory?

If yes, why?

Thanks in advance :)

**Review by leodido — COMMENTED** [review]
Left some comments

File: Cargo.toml (hunk 1 above)
Comment (squidboylan) [reply Cargo.toml:17]: I don't think it was mandatory, I did it because it seemed like a convenient time to update the version

**Pull request closed without merging**